src/Lecture1.hs

```diff
@@ -50,7 +51,8 @@ Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
 is 25.
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+sumOfSquares :: Num a => a -> a -> a
+sumOfSquares x y = x^2 + y^2
```

Comment:
This is the correct solution 👍🏻 I see that the Haskell compiler outputs some warnings here. This is because the operator `^` is polymorphic on both arguments and numeric literals are polymorphic as well. 

This is unfortunately 😞 But it's a problem when using the `^` operator. One of the possible solutions is to specify the type of constants 2 to some specific type using type annotations:

```suggestion
sumOfSquares x y = x ^ (2 :: Int) + y ^ (2 :: Int)
```

I agree that it looks a bit ugly now 😕 

Another approach is to use `*` in this particular case to avoid ambiguity:

```suggestion
sumOfSquares x y = x * x + y * y
```

Reply:
Changing into the first suggestion might make it less ugly by making the function `Int -> Int -> Int`, so you don't need to do the type annotation in the body itself, do you agree?

The second is actually my preferred 'human-linter' suggestion, given that it would keep the function bound to `Num` instead of `Int`.

Reply:
I haven't explained polymorphism yet but you can make an educated guess about the behaviour of `^` from its type signature:

```haskell
λ: :t (^)
(^) :: (Integral b, Num a) => a -> b -> a
```

Specifically, it allows the power to be of a different type than the base. So you still can have your function work with `Num` and use `Int` for `2` in `x ^ 2`